akka-stream/src/main/scala/akka/stream/io/SynchronousFileSource.scala

```diff
@@ -25,8 +25,7 @@ object SynchronousFileSource {
    * It materializes a [[Future]] containing the number of bytes read from the source file upon completion.
    */
   def apply(f: File, chunkSize: Int = DefaultChunkSize): Source[ByteString, Future[Long]] =
-    new Source(new SynchronousFileSource(f, chunkSize, DefaultAttributes, Source.shape("SynchronousFileSource")))
-      .named(DefaultAttributes.nameOption.get)
+    new Source(new SynchronousFileSource(f, chunkSize, DefaultAttributes, Source.shape("SynchronousFileSource")).wrap()) // TO DISCUSS: I had to add wrap() here to make the name available
```

Comment:
What was the exact problem? Anyway, named() called withAttributes() on the Source which called wrap().

> I fear that remembering where and when wrap() needs to be called will lead to very subtle bugs.

Yes. Calling wrap() does not change the semantics of the running stream though, it changes the module boundaries, i.e. modularity. The Layout cannot have any magical means to figure out this for you. You, as the designer of the user level DSL have the responsibility to choose the semantics that you want, and implement it properly. There are many possible semantics and the Layout cannot choose any of them. The Layout is just an immutable ADT for building hierarchical module networks, and it has only a few basic operations: grow, connect and wrap.